### PR TITLE
Fixes issue #3

### LINKED ISSUE #3: Implement dark theme
Switch to dark theme by clicking on the switch in navbar

### PR DESCRIPTION
**Pull request by Nikhil7174**
## Description

Added dark theme in the website

## Issue Resolved

Fixes #3.
Screenshot (125)
Screenshot (126)
Screenshot (127)
Screenshot (128)

## Screenshots

Provide some screenshots of the changes made (if applicable).

## Checklist

Please make sure to review the following before submitting your PR:

- [x] I have read the [contribution guidelines](https://github.com/ken1000minus7/Bloggy/blob/master/README.md).
- [x] I have read the code of conduct.
- [x] I have reviewed my submission in detail.

### CODE REVIEW

File: src/frontend/src/App.js [hunk 1]
```diff
@@ -8,30 +8,45 @@ import {LoginPage} from "./pages/LoginPage";
...
+import { Switch } from "@mui/material";
+import { createContext, useState } from "react";
+export const ThemeContext = createContext(null);
+
+
+=======
 import Footer from "./components/Footer";
 import { ToastContainer, toast } from 'react-toastify';
 import 'react-toastify/dist/ReactToastify.css';
+>>>>>>> 0e4db4609b5b994e912d40f97c04ced782ed0ec8
```
Comment (ken1000minus7) [inline src/frontend/src/App.js]: These are appearing due to merge conflicts, you need to fix this part of the code by removing the unnecessary lines and the repetitive imports

File: src/frontend/src/App.js [hunk 2]
```diff
@@ -8,30 +8,45 @@ import {LoginPage} from "./pages/LoginPage";
...
+    //     if((theme==="dark")!==event.target.checked){
+    //         if(theme==="dark"){
+    //             theme = "light"
+    //         }
+    //         else{
+    //             theme = "dark"
+    //         }
+    //         localStorage.setItem("theme",theme)
+    //     }
+    //     console.log(localStorage.getItem("theme"))
```
Comment (ken1000minus7) [inline src/frontend/src/App.js]: Since this code is no longer needed, you can just remove it instead of commenting

File: src/frontend/src/App.js [hunk 3]
```diff
@@ -8,30 +8,45 @@ import {LoginPage} from "./pages/LoginPage";
...
+    setTheme((curr) => (curr === "light" ? "dark" : "light"));
+        };
+    
     return (
-        <div className="App">
-            <Navbar changeTheme={changeTheme}/>
+        <ThemeContext.Provider value={{ theme, toggleTheme }}>
+        <div className="App" id={theme}>
+            <Navbar />
+            <Switch onChange={toggleTheme} checked={theme === "dark"} className="m-[10px] sm:m-0" />
```
Comment (ken1000minus7) [inline src/frontend/src/App.js]: You need to keep this switch in the navbar not here, you can pass the toggle theme as a prop to the navbar

**Review by ken1000minus7 — CHANGES_REQUESTED** [review]
Hey @Nikhil7174 the dark mode looks pretty great, although there r a few changes needed, i hv commented on the parts of the code that need to be fixed

Also just want to confirm, is the theme switching on clicking on the switch, since I can't confirm that from the screenshots

File: src/frontend/src/App.js (hunk 1 above)
Comment (ken1000minus7) [reply src/frontend/src/App.js]: Also this part of code is obvious to lead to compilation error but you have still pushed the code, I would recommend you to **ALWAYS** build and run the application locally and confirm that it's working fine before committing and pushing

Comment (Nikhil7174) [reply src/frontend/src/App.js]: I had run the program locally and then only pushed the code, somehow it had compiled successfully. But I will make the necessary changes to remove the conflict.

**Comment by Nikhil7174** [comment]
@ken1000minus7 The theme is switching by clicking on the swtich button.

File: src/frontend/src/components/footer.js [hunk 4]
```diff
@@ -72,7 +72,7 @@ const Footer = () => {
       </div>
     </div>
   </div>
-  <div className="text-center font-SourceSansPro p-6 bg-[#f0def4]">
+  <div className="text-center footerText2 font-SourceSansPro p-6 bg-[#f0def4]">
```
Comment (ken1000minus7) [inline src/frontend/src/components/footer.js]: The file is now renamed to `Footer.js` from `footer.js` so pull the fresh changes from master and make sure to add these changes in `Footer.js`

**Pull request closed without merging**